src/Symfony/Component/HttpKernel/DataCollector/ConfigDataCollector.php

```diff
@@ -175,9 +175,9 @@ public function getSymfonyMinorVersion()
     /**
      * Returns if the current Symfony version is a Long-Term Support one.
      */
-    public function isSymfonyLts(): bool
+    public function isSymfonyLts(): ?bool
     {
-        return $this->data['symfony_lts'];
+        return $this->data['symfony_lts'] ?? null;
```

Comment:
I would return `false` instead of `null`

Reply:
After having reviewed again the code, could there be a case to be made around leaving the method's signature as it is, and moving the assignment out of the `if` ?

Before the `if`, on line _47_, there is already a use of a constant of  the class `Kernel` (Plus, `Kernel` belongs to the same component as this class, so it will always exist):
```php
            'symfony_version' => Kernel::VERSION,
```

 There are some assignments inside the if which use `Kernel` as well, not the attribute `$kernel`.

```php
            $this->data['symfony_state'] = $this->determineSymfonyState();
            $this->data['symfony_minor_version'] = sprintf('%s.%s', Kernel::MAJOR_VERSION, Kernel::MINOR_VERSION);
            $this->data['symfony_lts'] = 4 === Kernel::MINOR_VERSION;
            $eom = \DateTime::createFromFormat('d/m/Y', '01/'.Kernel::END_OF_MAINTENANCE);
            $eol = \DateTime::createFromFormat('d/m/Y', '01/'.Kernel::END_OF_LIFE);
            $this->data['symfony_eom'] = $eom->format('F Y');
            $this->data['symfony_eol'] = $eol->format('F Y');
```

I would suggest bringing these assignments out of the `if`, as they do not depend on the optional attribute `$kernel` to exist. They depend on the `Kernel` class, which will always exist because it's in the same component.

Reply:
I'm fine with moving these outside of the "if" block.

Reply:
Changes made